airflow/jobs/scheduler_job_runner.py

```diff
@@ -2009,7 +2010,7 @@ def _find_zombies(self, *, session: Session) -> list[TI]:
             .join(DM, TI.dag_id == DM.dag_id)
             .where(
                 TI.state.in_((TaskInstanceState.RUNNING, TaskInstanceState.RESTARTING)),
-                TI.last_heartbeat_at < limit_dttm,
+                coalesce(TI.last_heartbeat_at, TI.updated_at) < limit_dttm,
```

Comment:
The downside with doing this is that update_at can be set for other reasons, it could be set by the scheduler for instance, so this _might_ in some circumstances keep a task alive longer than it should (but not indefinitely, it would like be a one-time only bump if at all)

🤔

Reply:
yeah agreed. i don't like the alternative which is to set `last_heartbeat_at` when we call `PATCH /run` though

Reply:
Oh huh. With the switch to storing `last_heartbeat_at` on TaskInstance (in 2.x zombie tasks are done by based on the LocalTaskJob row in the Job table), I don't think we ever clear it, which means there's a race condition where on resuming from deferral it could be instantly picked up as a zombie!